Presentations/CE-r-markdown.Rmd

```diff
@@ -0,0 +1,475 @@
+
+---
+
+# Including R code
+
+- R code in R Markdown goes inside **fenced code blocks**, as the one below
+
+````
+```{r}`r ''`
+# Your R code goes here
+```
+````
+
+---
+
+# Including R code
+
+- You can type the fenced code blocks directly or go to `Insert` > `R` in the script panel of RStudio to create a new block
+
+```{r echo = FALSE, out.width = "80%"}
+knitr::include_graphics("img/r-markdown/insert-new-r-fenced-block.png")
+```
+
+---
+
+# Including R code
+
+## Exercise 2
+
+Include the summary of a variable in the document.
+
+1. Create a header named `R Code` at the bottom of `markdown-template.Rmd`
+2. Create a new fenced code block where you load the sample dataset `mtcars`
+  + `mtcars` is a built-in dataset
+  + load it with: `data(mtcars)`
+3. Inside the same block, get the summary of the variable `mpg` with `summary(mtcars$mpg)`
+4. Knit the document to PDF and check how this is rendered in the resulting file
+
+---
+
+# Including R code
+
+```{r echo = FALSE, out.width = "80%"}
+knitr::include_graphics("img/r-markdown/r-markdown-r-code.png")
+```
+
+---
+
+# Including R code
+
+- Note that the output echoes both the code and the output
+- What if we wanted to include the output but not the code?
+- We use argument `echo = FALSE` in the fenced code block for that
+- Code block arguments are separated by commas inside the curly brackets, as in:
+
+````
+```{r, echo = FALSE}`r ''`
+data(mtcars)
+summary(mtcars$mpg)
+```
+````
+
+---
+
+# Including R code
+
+## Exercise 3
+
+Omit the code when knitting R code.
+
+1. Add the option `echo = FALSE` to the fenced code block created in exercise 2
+2. Knit the document into PDF and check the difference
+
+---
+
+# Including R code
+
+```{r echo = FALSE, out.width = "80%"}
+knitr::include_graphics("img/r-markdown/r-markdown-r-code-echo-F.png")
+```
+
+---
+
+# Including R code
+
+- To include R code but not the output, we can use the option `eval = FALSE`
+
+````
+```{r, eval = FALSE}`r ''`
+data(mtcars)
+summary(mtcars$mpg)
+```
+````
+
+---
+
+# Including R code
+
+```{r echo = FALSE, out.width = "80%"}
+knitr::include_graphics("img/r-markdown/r-markdown-r-code-eval-F.png")
+```
+
+---
+
+class: inverse, center, middle
+name: r-plots
+
+# R Plots
+
+<html><div style='float:left'></div><hr color='#D38C28' size=1px width=1100px></html>
+
+---
+
+# Including R plots
+
+- Adding R plots is similar to adding R code
+- Include the code producing the plot in a fenced block
+- The block option `echo = FALSE` is useful when we only want to include the plot but not the code producing it
+
+---
+
+# Including R plots
+
+## Exercise 4
+
+Include an R plot in your document.
+
+1. Create a new header named `R Plots` at the bottom of `markdown-template.Rmd`
+2. Create a new fenced code block with the option `echo` set to `FALSE`
+3. Add the following code inside the new block:
+
+```{r, eval = FALSE}
+plot(mtcars$wt,
```

Comment:
@luizaandrade with this plot I wanted to keep a balance between showing something that is not complicated so the data vis code syntax doesn't distract from R Markdown and including a plot not distractingly ugly, while not having to import an external library (like `ggplot2`) for it

Reply:
That seems fine to me. I'd mention it during the presentation, though: just say that if they have taken our R course, they probably know this is not our preferred way of making plots, but that for this session we only wanted to use base R so we don't need to get into discussing packages.